### PR TITLE
Add support for named slots

### PR DESCRIPTION
**Pull request by natemoo-re**
## Changes

Previously, `.astro` files only supported a single default `<slot>`. This PR brings Astro's `<slot>` support inline with the HTML spec. It adds support for multiple named slots and allows `<slots>` to render fallback content when they aren't passed any children.

## Testing

Added `astro-slot` tests

## Docs

✅ Docs updated

### CODE REVIEW

File: .changeset/fair-flowers-sleep.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+'astro': minor
```
Comment (matthewp) [inline .changeset/fair-flowers-sleep.md:2]: This definitely seems non-breaking to me. I think we should avoid forcing people to update their package.json version numbers as much as possible, and this seems like a clear case of added capability.

Comment (natemoo-re) [reply .changeset/fair-flowers-sleep.md:2]: Sweet! I'm good with that.

**Comment by natemoo-re** [comment]
@matthewp updated to use a `patch` version but I suspect this will be released with `astro@0.18.0` since we have some `minor` PRs that are very close.

**Comment by matthewp** [comment]
@natemoo-re I have a `minor` change too, so I think that's ok if we wait and have a "big release".

**Pull request merged**